core/src/main/java/com/linecorp/armeria/server/ServiceErrorHandler.java

```diff
@@ -0,0 +1,154 @@
+     *
+     * @return an {@link HttpResponse} to send to the client, or {@code null} to let the next handler
+     *         specified with {@link #orElse(ServiceErrorHandler)} handle the event.
+     */
+    @Nullable
+    HttpResponse onServiceException(ServiceRequestContext ctx, Throwable cause);
+
+    /**
+     * Returns an {@link AggregatedHttpResponse} generated from the given {@link HttpStatus}, {@code message}
+     * and {@link Throwable}. When {@code null} is returned, the next {@link ServerErrorHandler}
+     * in the invocation chain will be used as a fallback (See {@link #orElse(ServiceErrorHandler)}
+     * for more information).
+     *
+     * @param config the {@link ServiceConfig} that provides the configuration properties.
+     * @param headers the received {@link RequestHeaders}, or {@code null} in case of severe protocol violation.
+     * @param status the desired {@link HttpStatus} of the error response.
+     * @param description an optional human-readable description of the error.
+     * @param cause an optional exception that may contain additional information about the error, such as
+     *              {@link ContentTooLargeException}.
+     *
+     * @return an {@link AggregatedHttpResponse}, or {@code null} to let the next handler specified with
+     *         {@link #orElse(ServiceErrorHandler)} handle the event.
+     */
+    @Nullable
+    default AggregatedHttpResponse renderStatus(ServiceConfig config,
+                                                @Nullable RequestHeaders headers,
+                                                HttpStatus status,
+                                                @Nullable String description,
+                                                @Nullable Throwable cause) {
+        return null;
+    }
+
+    /**
+     * Returns a newly created {@link ServiceErrorHandler} that tries this {@link ServiceErrorHandler} first and
+     * then the specified {@link ServiceErrorHandler} when the first call returns {@code null}.
+     *
+     * <pre>{@code
+     * ServiceErrorHandler handler = (ctx, cause) -> {
+     *     if (cause instanceof FirstException) {
+     *         return HttpResponse.of(200);
+     *     }
+     *     return null;
+     * }
+     * assert handler.onServiceException(ctx, new FirstException()) != null;
+     * assert handler.onServiceException(ctx, new SecondException()) == null;
+     * assert handler.onServiceException(ctx, new ThirdException()) == null;
+     *
+     * ServiceErrorHandler combinedHandler = handler.orElse((ctx, cause) -> {
+     *     if (cause instanceof SecondException) {
+     *         return HttpResponse.of(200);
+     *     }
+     *     return null;
+     * });
+     *
+     * assert handler.onServiceException(ctx, new FirstException()) != null;
+     * assert handler.onServiceException(ctx, new SecondException()) != null;
+     * assert handler.onServiceException(ctx, new ThirdException()) == null;
```

Comment:
the javadocs in `ServerErrorHandler` are also incorrect 😅    Do you mind also fixing them?

```suggestion
     * assert combinedHandler.onServiceException(ctx, new FirstException()) != null;
     * assert combinedHandler.onServiceException(ctx, new SecondException()) != null;
     * assert combinedHandler.onServiceException(ctx, new ThirdException()) == null;
```